homeassistant/core.py

```diff
@@ -333,6 +334,9 @@ def _async_stop_handler(self, *args):
     @callback
     def _async_restart_handler(self, *args):
         """Restart Home Assistant."""
+        if subprocess.call([sys.argv[0], '--script', 'check_config']):
+            _LOGGER.error("check_config failed. Not restarting.")
```

Comment:
You will need to create a persistent notification to notify the user or they won't know.